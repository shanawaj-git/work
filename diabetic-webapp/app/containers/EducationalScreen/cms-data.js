const educationl1 = require('images/educationImage1.svg');
const educationl2 = require('images/educationImage2.svg');
const education3 = require('images/educationImage3.svg');

export const cmsEducationalData = {
  data: [
    {
      id: 1,
      attributes: {
        header_text: '<p>Living with <strong>Diabetes</strong></p>',
        description:
          '<p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Donec odio. Quisque volutpat mattis eros. Nullam malesuada erat ut turpis. Suspendisse urna nibh, viverra non, semper suscipit, posuere a, pede.</span></p>',
        createdAt: '2022-08-17T06:57:57.538Z',
        updatedAt: '2022-08-17T06:58:42.674Z',
        publishedAt: '2022-08-17T06:58:42.668Z',
        header_img: {
          data: [
            {
              id: 1,
              attributes: {
                name: 'educational1.svg',
                alternativeText: 'educational1.svg',
                caption: 'educational1.svg',
                width: 289,
                height: 312,
                formats: null,
                hash: 'educational1_9b5a4acd55',
                ext: '.svg',
                mime: 'image/svg+xml',
                size: 3458.48,
                url: '/uploads/educational1_9b5a4acd55.svg',
                imageSrc: educationl1,
                previewUrl: null,
                provider: 'local',
                provider_metadata: null,
                createdAt: '2022-08-17T06:32:46.340Z',
                updatedAt: '2022-08-17T06:32:46.340Z',
              },
            },
          ],
        },
      },
    },
    {
      id: 2,
      attributes: {
        header_text:
          '<p>Understanding your <strong>medication and treatment</strong></p>',
        description:
          '<ul><li>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</li><li>Aliquam tincidunt mauris eu risus.</li><li>Vestibulum auctor dapibus neque.</li><li>Nunc dignissim risus id metus.</li></ul>',
        createdAt: '2022-08-17T07:49:08.707Z',
        updatedAt: '2022-08-17T07:49:09.533Z',
        publishedAt: '2022-08-17T07:49:09.530Z',
        header_img: {
          data: [
            {
              id: 1,
              attributes: {
                name: 'educational1.svg',
                alternativeText: 'educational1.svg',
                caption: 'educational1.svg',
                width: 289,
                height: 312,
                formats: null,
                hash: 'educational1_9b5a4acd55',
                ext: '.svg',
                mime: 'image/svg+xml',
                size: 3458.48,
                url: '/uploads/educational1_9b5a4acd55.svg',
                imageSrc: educationl2,
                previewUrl: null,
                provider: 'local',
                provider_metadata: null,
                createdAt: '2022-08-17T06:32:46.340Z',
                updatedAt: '2022-08-17T06:32:46.340Z',
              },
            },
          ],
        },
      },
    },
    {
      id: 3,
      attributes: {
        header_text:
          '<p>When should I take <strong>the medication?</strong></p>',
        description:
          '<p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">It is a long-established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using \'Content here, content here\', making it look like readable</span></p>',
        createdAt: '2022-08-17T11:02:16.899Z',
        updatedAt: '2022-08-17T11:09:52.095Z',
        publishedAt: '2022-08-17T11:02:32.420Z',
        header_img: {
          data: [
            {
              id: 1,
              attributes: {
                name: 'educational1.svg',
                alternativeText: 'educational1.svg',
                caption: 'educational1.svg',
                width: 289,
                height: 312,
                formats: null,
                hash: 'educational1_9b5a4acd55',
                ext: '.svg',
                mime: 'image/svg+xml',
                size: 3458.48,
                url: '/uploads/educational1_9b5a4acd55.svg',
                imageSrc: education3,
                previewUrl: null,
                provider: 'local',
                provider_metadata: null,
                createdAt: '2022-08-17T06:32:46.340Z',
                updatedAt: '2022-08-17T06:32:46.340Z',
              },
            },
          ],
        },
      },
    },
    {
      id: 4,
      attributes: {
        header_text:
          '<p>Lifestyle changes to <strong>control your diabetes</strong></p>',
        description:
          '<ul><li>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</li><li>Aliquam tincidunt mauris eu risus.</li></ul><p><br>&nbsp;</p>',
        createdAt: '2022-08-17T11:03:05.331Z',
        updatedAt: '2022-08-17T11:10:00.638Z',
        publishedAt: '2022-08-17T11:03:06.326Z',
        header_img: {
          data: [
            {
              id: 1,
              attributes: {
                name: 'educational1.svg',
                alternativeText: 'educational1.svg',
                caption: 'educational1.svg',
                width: 289,
                height: 312,
                formats: null,
                hash: 'educational1_9b5a4acd55',
                ext: '.svg',
                mime: 'image/svg+xml',
                size: 3458.48,
                url: '/uploads/educational1_9b5a4acd55.svg',
                imageSrc: education3,
                previewUrl: null,
                provider: 'local',
                provider_metadata: null,
                createdAt: '2022-08-17T06:32:46.340Z',
                updatedAt: '2022-08-17T06:32:46.340Z',
              },
            },
          ],
        },
      },
    },
  ],
  meta: {
    pagination: {
      page: 1,
      pageSize: 25,
      pageCount: 1,
      total: 4,
    },
  },
};
