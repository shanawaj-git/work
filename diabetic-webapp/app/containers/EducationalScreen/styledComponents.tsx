import styled from 'styled-components';
import { Carousel } from '@albathanext/design-system';

export const CarouselImage = styled.div`
  direction: rtl;
`;

export const CarouselHeader = styled.div`
  display: block;
  padding: 10px 30px;
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-weight: 500;
  font-size: 32px;
`;

export const CarouselDescription = styled.div`
  padding: 10px 30px;
  font-family: 'Neue Haas Grotesk Display Pro';
  font-style: normal;
  font-weight: 500;
  font-size: 15px;
  line-height: 22px;
  letter-spacing: 0.3px;
  color: #495151;
`;

export const CarouselNextButton = styled.button`
  pointer-events: all;
  margin-top: 60px;
  margin-right: 5px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0px 10px;
  width: 159.5px;
  height: 48px;
  /* BG/Primary */
  background: #00664f;
  border-radius: 12px;
  /* Inside auto layout */
  flex: none;
  order: 1;
  flex-grow: 1;
`;

export const CarouselSkipButton = styled.button`
  pointer-events: all;
  margin-top: 60px;
  border-width: 0;
  font-family: 'system-ui';
  text-decoration: underline;
  font-style: normal;
  font-weight: 600;
  font-size: 16px;
  line-height: 20px;
  display: flex;
  align-items: center;
  text-align: center;
  letter-spacing: 0.4px;
  color: #000000;
  flex: none;
  order: 0;
  flex-grow: 0;
`;

export const EducationalCarousel = styled(Carousel)`
  display: flex;
  flex-direction: column;

  menu,
  ol,
  ul {
    padding: 0px 10px;
    list-style-image: url("data:image/svg+xml,%3Csvg width='14' height='14' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Ccircle opacity='.2' cx='7' cy='7' r='7' fill='%2312A1C0'/%3E%3Ccircle cx='7' cy='7' r='3' fill='%2312A1C0'/%3E%3C/svg%3E");
    font-family: 'Neue Haas Grotesk Display Pro';
    font-style: normal;
    font-weight: 600;
    font-size: 15px;
    line-height: 22px;
    letter-spacing: 0.3px;
    color: #495151;
  }

  .MuiButton-root {
    border: none;
    background: transparent !important;
    color: white !important;
    justify-content: left;
  }
  .MuiButton-root:hover {
    color: transparent;
    box-shadow: none;
  }
  .MuiMobileStepper-dots {
    margin-bottom: 40px;
    margin-left: -45vw;
  }
  .MuiMobileStepper-dot,
  .MuiMobileStepper-dot {
    width: 8px;
    height: 4px;
    /* Stroke/Inactive */
    background: #d4d9d9;
    border-radius: 10px;
    /* Inside auto layout */
    flex: none;
    order: 1;
    flex-grow: 0;
  }

  .MuiMobileStepper-dotActive {
    width: 53px;
    height: 4px;
    background: #12a1c0;
    border-radius: 10px;
    flex: none;
    order: 1;
    flex-grow: 0;
  }
`;
