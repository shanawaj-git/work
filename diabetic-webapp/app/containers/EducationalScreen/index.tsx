import React, { useState } from 'react';
import PropTypes, { any, InferProps } from 'prop-types';
import { injectIntl } from 'react-intl';
import ReactHtmlParser from 'react-html-parser';
import { cmsEducationalData } from './cms-data';
import {
  CarouselDescription,
  CarouselHeader,
  CarouselImage,
  CarouselNextButton,
  CarouselSkipButton,
  EducationalCarousel,
} from './styledComponents';

const NEW_PRESCRIPTION_SCREEN = 'prescription-upload';
function EducationalScreen({
  intl,
  history,
  location,
}: EducationalScreenTypes) {
  const [currentIndex, setCurrentIndex] = useState(0);
  const caroselData = [];
  const messages = {
    skipButton: intl.formatMessage({
      id: 'skipButton.title',
      defaultMessage: 'Skip',
    }),
    nextButton: intl.formatMessage({
      id: 'nextButton.title',
      defaultMessage: 'Next',
    }),
  };

  function onSkipButtonClick(event) {
    event.stopPropagation();
    event.preventDefault();
    history.push({
      pathname: NEW_PRESCRIPTION_SCREEN,
      state: {},
    });
    return false;
  }

  function onNextButtonClick() {
    if (currentIndex === caroselData.length - 1) {
      history.push({
        pathname: NEW_PRESCRIPTION_SCREEN,
        state: {},
      });
    }
  }

  function onIndexChanged(index) {
    setCurrentIndex(index);
  }

  cmsEducationalData.data.forEach(cmsData => {
    caroselData.push(
      <>
        <CarouselImage>
          <img
            src={cmsData.attributes.header_img.data[0].attributes.imageSrc}
            alt="header_img"
          />
        </CarouselImage>
        <CarouselHeader>
          {ReactHtmlParser(cmsData.attributes.header_text)}
        </CarouselHeader>
        <CarouselDescription>
          {ReactHtmlParser(cmsData.attributes.description)}
        </CarouselDescription>
      </>,
    );
  });

  return (
    <>
      <EducationalCarousel
        content={caroselData}
        position="right"
        maxSteps={caroselData.length}
        direction="ltr"
        onIndexChanged={onIndexChanged}
        backButton
        nextButton
        nextIcon={
          <CarouselNextButton onClick={onNextButtonClick}>
            {messages.nextButton}
          </CarouselNextButton>
        }
        backIcon={
          <CarouselSkipButton onClick={onSkipButtonClick}>
            {messages.skipButton}
          </CarouselSkipButton>
        }
      />
    </>
  );
}

const EducationalScreenPropsTypes = {
  intl: any,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }),
  location: PropTypes.shape({
    search: PropTypes.string.isRequired,
  }),
};

type EducationalScreenTypes = InferProps<typeof EducationalScreenPropsTypes>;

EducationalScreen.propTypes = EducationalScreenPropsTypes;

export default injectIntl(EducationalScreen);
