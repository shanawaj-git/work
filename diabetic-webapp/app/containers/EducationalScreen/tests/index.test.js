import React from 'react';

import { fireEvent, render } from 'react-testing-library';
import { act } from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import EducationalScreen from '../index';

const PRESCRIPTION_UPLOAD = { pathname: 'prescription-upload', state: {} };
jest.mock('react-router-dom', () => ({
  __esModule: true,
  useLocation: jest.fn().mockReturnValue({
    pathname: '/educational-screen',
    search: '',
    hash: '',
    state: null,
    key: '',
  }),
}));

describe('<EducationalScreen/>', () => {
  it('should render and match the snapshot', () => {
    const mockLocation = {
      search: '',
    };

    const props = {
      location: mockLocation,
    };

    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale="en">
        <EducationalScreen {...props} />
      </IntlProvider>,
    );

    expect(firstChild).toMatchSnapshot();
  });

  it('should skip the carousel if the skip button is clicked', () => {
    const mockHistory = {
      push: jest.fn(),
    };

    const mockLocation = {
      search: '',
    };

    const props = {
      location: mockLocation,
    };

    act(() => {
      render(
        <IntlProvider locale="en">
          <EducationalScreen history={mockHistory} {...props} />
        </IntlProvider>,
      );
      const skipButton = document.querySelector('.Mui-disabled > div > button');
      fireEvent.click(
        skipButton,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );
      expect(mockHistory.push).toBeCalledWith(PRESCRIPTION_UPLOAD);
    });
  });

  it('should render the next content of carousel when Next button is clicked', () => {
    const mockHistory = {
      push: jest.fn(),
    };

    const mockLocation = {
      search: '',
    };

    const props = {
      location: mockLocation,
    };

    act(() => {
      render(
        <IntlProvider locale="en">
          <EducationalScreen history={mockHistory} {...props} />
        </IntlProvider>,
      );

      const nextButton = document.querySelector(
        '.MuiMobileStepper-root > button:nth-child(3)',
      );

      fireEvent.click(
        nextButton,
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        }),
      );

      const caroselComponet = document.querySelector(
        '.react-swipeable-view-container',
      );

      expect(caroselComponet).toBeDefined();
    });
  });
});
