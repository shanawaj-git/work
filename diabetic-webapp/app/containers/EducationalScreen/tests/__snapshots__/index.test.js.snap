// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EducationalScreen/> should render and match the snapshot 1`] = `
<div
  class="sc-EHOje liupJE"
>
  <div
    style="overflow-x: hidden; flex: 1; flex-direction: flex; height: 100%;"
  >
    <div
      class="react-swipeable-view-container"
      style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
    >
      <div
        aria-hidden="false"
        data-swipeable="true"
        style="width: 100%; flex-shrink: 0; overflow: auto;"
      >
        <div
          class="sc-bdVaJa kgLnZe"
        >
          <img
            alt="header_img"
            src="IMAGE_MOCK"
          />
        </div>
        <div
          class="sc-bwzfXH kPCRcv"
        >
          <p>
            Living with 
            <strong>
              Diabetes
            </strong>
          </p>
        </div>
        <div
          class="sc-htpNat eYTBro"
        >
          <p>
            <span
              style="background-color: rgb(255, 255, 255); color: rgb(0, 0, 0);"
            >
              Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Donec odio. Quisque volutpat mattis eros. Nullam malesuada erat ut turpis. Suspendisse urna nibh, viverra non, semper suscipit, posuere a, pede.
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiMobileStepper-root MuiMobileStepper-positionRight css-rh92k-MuiPaper-root-MuiMobileStepper-root"
  >
    <button
      class="root StyledButton-sc-1apep4b-0 bXswYh MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="back-btn"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <div
        style="visibility: visible;"
      >
        <button
          class="sc-ifAKCX SZkMy"
        >
          Skip
        </button>
      </div>
    </button>
    <div
      class="MuiMobileStepper-dots css-1384nzf-MuiMobileStepper-dots"
    >
      <div
        class="MuiMobileStepper-dot MuiMobileStepper-dotActive css-26w9jf-MuiMobileStepper-dot"
      />
      <div
        class="MuiMobileStepper-dot css-114p2tk-MuiMobileStepper-dot"
      />
      <div
        class="MuiMobileStepper-dot css-114p2tk-MuiMobileStepper-dot"
      />
      <div
        class="MuiMobileStepper-dot css-114p2tk-MuiMobileStepper-dot"
      />
    </div>
    <button
      class="root StyledButton-sc-1apep4b-0 bXswYh MuiButton-root text MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      data-testid="next-btn"
      tabindex="0"
      type="button"
    >
      <div
        style="visibility: visible;"
      >
        <button
          class="sc-bxivhb eZnAXq"
        >
          Next
        </button>
      </div>
    </button>
  </div>
</div>
`;
